src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java

```diff
@@ -37,61 +51,25 @@ public List<OrderDetails> getAllOrders() {
 
         List<OrderDetails> list = new ArrayList<>(orders.size());
         for (Order order : orders) {
-            OrderDetails orderDetails = collectOrderDetails(order);
-            list.add(orderDetails);
+            list.add(collectOrderDetails(order));
         }
         return list;
     }
 
-
-    private OrderDetails collectOrderDetails(Order order) {
-        OrderDetails orderDetails = new OrderDetails();
-        orderDetails.setId(order.getId());
+    private OrderDetails collectOrderDetails(Order order){
+        OrderDetails orderDetails = orderMapper.toOrderDetails(order);
         orderDetails.setOrderItems(collectOrderItemDetails(order.getId()));
-        orderDetails.setCustomerId(order.getUserId());
-        orderDetails.setUpdatedAt(order.getUpdatedAt());
-        orderDetails.setCreatedAt(order.getCreatedAt());
         orderDetails.setTotalPrice(getTotalPriceForOrder(orderDetails.getOrderItems()));
         return orderDetails;
     }
 
-    private BigDecimal getTotalPriceForOrder(List<OrderItemDetails> orderItems) {
-        BigDecimal result = BigDecimal.ZERO;
-        for (OrderItemDetails orderItem : orderItems) {
-            BigDecimal totalPrice = orderItem.getProduct().getPrice()
-                    .multiply(BigDecimal.valueOf(orderItem.getQuantity()));
-            result = result.add(totalPrice);
-        }
-        return result;
-    }
-
     private List<OrderItemDetails> collectOrderItemDetails(Integer orderId) {
-        return orderItemRepository.getOrderItemsByOrderId(orderId).stream()
-                .map(orderItem -> convertOrderItemToOrderItemDetails(orderItem))
-                .toList();
+        List<OrderItem> orderItemsByOrderId = orderItemRepository.getOrderItemsByOrderId(orderId);
+        List<Integer> productIds = orderItemsByOrderId.stream().map(OrderItem::getProductId).toList();
+        Map<Integer, Product> productsByIds = productRepository.getProductsByIds(productIds);
+        return orderItemsByOrderId.stream().map(orderItem -> orderItemMapper
+                .toOrderItemDetails(orderItem,productMapper.toProductDetails(productsByIds
+                        .get(orderItem.getProductId())))).toList();
```

Comment:
your code will be a bit more readable if you put each sream operation in new line. also if you extract product details in a separate variable your code will be more readable
smth like:
```java
        return orderItemsByOrderId.stream()
                .map(orderItem -> {
                    ProductDetails pd = productMapper.toProductDetails(productsByIds.get(orderItem.getProductId()));
                    return orderItemMapper.toOrderItemDetails(orderItem, pd);
                })
                .toList();

Reply:
yes i did it thanks